src/ftest.jl

```diff
@@ -0,0 +1,75 @@
+type FTestResult
+    mod1::RegressionModel
+    mod2::RegressionModel
+    SSR1::Float64 # sum of squared residuals, model 1
+    SSR2::Float64 # sum of squared residuals, model 2
+    df1::Int # degrees of freedom, model 1 residuals
+    df2::Int # degrees of freedom, model 2 residuals
+    MSR1::Float64 # mean of squared residuals, model 1
+    MSR2::Float64 # mean of squared residuals, model 2
+    fstat::Float64 # f statistic
+    pval::Float64 # p value
+end
+
+type MultiFTestResult
+    results::Array{FTestResult, 1}
+end
+
+function ftest(mod1::RegressionModel, mod2::RegressionModel)
+    SSR1 = deviance(mod1.model.rr)
+    SSR2 = deviance(mod2.model.rr)
+
+    nparams1 = length(mod1.model.pp.beta0)
+    nparams2 = length(mod2.model.pp.beta0)
+
+    df2 = nparams1-nparams2
+    df1 = dof_residual(mod1)
+
+    MSR1 = (SSR2-SSR1)/df2
+    MSR2 = SSR1/df1
+
+    fstat = MSR1/MSR2
+    pval = ccdf(FDist(df2, df1), fstat)
+
+    return FTestResult(mod1, mod2, SSR1, SSR2, df1, df2, MSR1, MSR2, fstat, pval)
+end
+
+function ftest(mods::RegressionModel...)
+    nmodels = length(mods)
+    results = Array{FTestResult, 1}((nmodels^2)-nmodels)
```

Comment:
What R's `anova` and Stata's `nestreg` do is to compare each model to the previous one. Computing all possible comparisons quickly gets very messy.